pipelines/LSSTComCamSim/nightly-validation.yaml

```diff
@@ -12,7 +12,6 @@ tasks:
   analyzePreSourceTableCore:
     class: lsst.analysis.tools.tasks.SourceTableVisitAnalysisTask
     config:
-      connections.data: preSourceTable_visit
       connections.inputName: preSourceTable_visit
```

Comment:
I don't think we need to provide `data` here, right? Here is where the connections are defined for this task:
https://github.com/lsst/analysis_tools/blob/e452f7d79f336b7d8be42b4ac7a24ac7647eb724/python/lsst/analysis/tools/tasks/sourceTableVisitAnalysis.py#L35
```python
class SourceTableVisitAnalysisConnections(
    AnalysisBaseConnections,
    dimensions=("visit", "band"),
    defaultTemplates={"inputName": "sourceTable_visit"},
):
    data = ct.Input(
        doc="Visit based source table to load from the butler",
        name="sourceTable_visit",
        storageClass="ArrowAstropy",
        dimensions=("visit", "band"),
        deferLoad=True,
    )

    camera = ct.PrerequisiteInput(
        doc="Input camera to use for focal plane geometry.",
        name="camera",
        storageClass="Camera",
        dimensions=("instrument",),
        isCalibration=True,
    )
```

Reply:
Changed the connection class to the following:
```python
class SourceTableVisitAnalysisConnections(
    AnalysisBaseConnections,
    dimensions=("visit", "band"),
    defaultTemplates={"inputName": "sourceTable_visit"},
):
    data = ct.Input(
        doc="Visit based source table to load from the butler",
        name="{inputName}",  # Changed this line!
        storageClass="ArrowAstropy",
        dimensions=("visit", "band"),
        deferLoad=True,
    )

    camera = ct.PrerequisiteInput(
        doc="Input camera to use for focal plane geometry.",
        name="camera",
        storageClass="Camera",
        dimensions=("instrument",),
        isCalibration=True,
    )
```

Reply:
Right, the only reason we'd need to define data is if it was involved in a contract at the end of the pipeline yaml.